framework/src/modules/poa/commands/update_authority.ts

```diff
@@ -0,0 +1,159 @@
+	MESSAGE_TAG_POA,
+	EMPTY_BYTES,
+	UpdateAuthority,
+} from '../constants';
+import {
+	CommandExecuteContext,
+	CommandVerifyContext,
+	VerificationResult,
+	VerifyStatus,
+} from '../../../state_machine';
+import { UpdateAuthorityValidatorParams } from '../types';
+import { ChainPropertiesStore, SnapshotStore, ValidatorStore } from '../stores';
+import { ValidatorsMethod } from '../../pos/types';
+import { AuthorityUpdateEvent } from '../events/authority_update';
+
+export class UpdateAuthorityCommand extends BaseCommand {
+	public schema = updateAuthorityValidatorSchema;
+	private _validatorsMethod!: ValidatorsMethod;
+
+	public get name(): string {
+		return COMMAND_UPDATE_AUTHORITY;
+	}
+
+	public addDependencies(validatorsMethod: ValidatorsMethod) {
+		this._validatorsMethod = validatorsMethod;
+	}
+
+	public async verify(
+		context: CommandVerifyContext<UpdateAuthorityValidatorParams>,
+	): Promise<VerificationResult> {
+		const { newValidators, threshold, validatorsUpdateNonce } = context.params;
+		try {
+			validator.validate(updateAuthorityValidatorSchema, context.params);
+		} catch (err) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: err as Error,
+			};
+		}
+		if (newValidators.length < 1 || newValidators.length > MAX_NUM_VALIDATORS) {
+			throw new Error('Invalid number of newValidators.');
+		}
+		if (
+			!objectUtils.bufferArrayOrderByLex(newValidators.map(newValidator => newValidator.address))
+		) {
+			throw new Error('Addresses in newValidators are not lexicographical ordered.');
+		}
+
+		let totalWeight = BigInt(0);
+		for (const newValidator of newValidators) {
+			const validatorExists = await this.stores
+				.get(ValidatorStore)
+				.has(context, newValidator.address);
+			if (!validatorExists) {
+				throw new Error(
+					`${address.getLisk32AddressFromPublicKey(
+						newValidator.address,
+					)} is not a valid validator.`,
+				);
+			}
+			totalWeight += newValidator.weight;
+		}
+
+		if (totalWeight > MAX_UINT64) {
+			throw new Error('TotalWeight out of range.');
+		}
+
+		if (threshold < totalWeight / BigInt(3) + BigInt(1) || threshold > totalWeight) {
+			throw new Error('Invalid threshold.');
+		}
+
+		const chainPropertiesStore = await this.stores
+			.get(ChainPropertiesStore)
+			.get(context, EMPTY_BYTES);
+		if (validatorsUpdateNonce !== chainPropertiesStore.validatorsUpdateNonce) {
+			throw new Error('Invalid validatorsUpdateNonce.');
+		}
+
+		return {
+			status: VerifyStatus.OK,
+		};
+	}
+
+	public async execute(
+		context: CommandExecuteContext<UpdateAuthorityValidatorParams>,
+	): Promise<void> {
+		const { newValidators, threshold, validatorsUpdateNonce, aggregationBits, signature } =
+			context.params;
+		const message = codec.encode(validatorSignatureMessageSchema, {
+			newValidators,
+			threshold,
+			validatorsUpdateNonce,
+		});
+		const validatorInfos = [];
+		const snapshotStore = await this.stores.get(SnapshotStore).get(context, Buffer.from([0]));
```

Comment:
I think you need to use `intToBuffer` method from cryptography.utils here for the 4 byte key

```suggestion
		const snapshotStore = await this.stores.get(SnapshotStore).get(context, intToBuffer(0, 4));
```